### ISSUE TITLE
No way to access connection rejection promise when auto retry after disconnect in provider

### ISSUE THREAD

**Issue by deeg** [ueberdosis/hocuspocus#529]
**Description**

**Steps to reproduce the bug**
1. Be using HP Provider
2. Go offline (Turn WIFI off)
3. Come back online (Turn WIFI on)
4. HP Provider tries to re-connect
5. After max amount rejection promise is swallowed and user not able to determine retry is over

**Expected behavior**
There is some how access to promise rejection of connect so user knows when retries are over and can take additional action

**Screenshot, video, or GIF**
This is the code where connect is called and rejection promise not handled: https://github.com/ueberdosis/hocuspocus/blob/476898bfb892227ea9fbd18433e01e9717b9d549/packages/provider/src/HocuspocusProvider.ts#L616

**Environment?**
 - operating system: Mac
 - browser: Chrome
 - mobile/desktop: Desktop
 - hocuspocus version: Provider 1.0.0

**Comment by iyk2h** [comment]
Hello,

I was wondering if this issue has been resolved or if there are any updates regarding it. If not, are there any workarounds available?

Thank you for your time and help!

**Comment by grootgordon** [comment]
> Hello,
> 
> I was wondering if this issue has been resolved or if there are any updates regarding it. If not, are there any workarounds available?
> 
> Thank you for your time and help!

I stucked this issue in `iOS15/iOS16`. 
TL;DR: 
`setInterval` or  `setTimeout` was unreliable in iOS15/iOS16 webkit, i use `requestAnimationFrame` to replace it.

you can refer this issue for more detail: https://github.com/ueberdosis/hocuspocus/issues/661#issuecomment-1659529653
